subprojects/ide-native/src/main/java/org/gradle/ide/xcode/internal/xcodeproj/PBXReference.java

```diff
@@ -0,0 +1,161 @@
+    @Nullable
+    private String path;
+    /**
+     * The "base" path of the reference. The absolute path is resolved by prepending the resolved
+     * base path.
+     */
+    private SourceTree sourceTree;
+
+    public PBXReference(String name, @Nullable String path, SourceTree sourceTree) {
+        this.name = Preconditions.checkNotNull(name);
+        this.path = path;
+        this.sourceTree = Preconditions.checkNotNull(sourceTree);
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Nullable
+    public String getPath() {
+        return path;
+    }
+
+    public void setPath(String v) {
+        path = v;
+    }
+
+    public SourceTree getSourceTree() {
+        return sourceTree;
+    }
+
+    public void setSourceTree(SourceTree v) {
+        sourceTree = v;
+    }
+
+    @Override
+    public String isa() {
+        return "PBXReference";
+    }
+
+    @Override
+    public int stableHash() {
+        return name.hashCode();
+    }
+
+    @Override
+    public void serializeInto(XcodeprojSerializer s) {
+        super.serializeInto(s);
+
+        s.addField("name", name);
+        if (path != null) {
+            s.addField("path", path);
+        }
+        s.addField("sourceTree", sourceTree.toString());
+    }
+
+    @Override
+    public String toString() {
+        return String.format(
+            "%s name=%s path=%s sourceTree=%s",
+            super.toString(),
+            getName(),
+            getPath(),
+            getSourceTree());
+    }
+
+    public enum SourceTree {
+        /**
+         * Relative to the path of the group containing this.
+         */
+        GROUP("<group>"),
+
+        /**
+         * Absolute system path.
+         */
+        ABSOLUTE("<absolute>"),
+        /**
+         * Relative to the build setting {@code BUILT_PRODUCTS_DIR}.
+         */
+        BUILT_PRODUCTS_DIR("BUILT_PRODUCTS_DIR"),
+
+        /**
+         * Relative to the build setting {@code SDKROOT}.
+         */
+        SDKROOT("SDKROOT"),
+
+        /**
+         * Relative to the directory containing the project file {@code SOURCE_ROOT}.
+         */
+        SOURCE_ROOT("SOURCE_ROOT"),
+
+        /**
+         * Relative to the Developer content directory inside the Xcode application
+         * (e.g. {@code /Applications/Xcode.app/Contents/Developer}).
+         */
+        DEVELOPER_DIR("DEVELOPER_DIR"),;
+
+        private final String rep;
+
+        SourceTree(String str) {
+            rep = str;
+        }
+
+        /**
+         * Return a sourceTree given a build setting that is typically used as a source tree prefix.
+         *
+         * The build setting may be optionally prefixed by '$' which will be stripped.
+         */
+        public static Optional<SourceTree> fromBuildSetting(String buildSetting) {
+            String data = CharMatcher.is('$').trimLeadingFrom(buildSetting);
+            if (data.equals("BUILT_PRODUCTS_DIR")) {
```

Comment:
I have always been a bit confused by the JDK limitation. If I understand correctly, only the code that needs to run to give meaningful error should support JDK 5, everything else should be fine to use JDK 1.7. Given the `ide-native` support will run as part of Gradle doing actual work (aka it has passed the check for >= JDK 1.7), we could force this project with JDK 1.7 compatibility.